docs/resources/task.md

```diff
@@ -0,0 +1,224 @@
+
+- `{machine}` - Any [EC2 instance type](https://aws.amazon.com/ec2/instance-explorer) (e.g. `g4dn.xlarge`).
+
+#### Google Cloud Platform
+
+- `{machine}` - Any [GCP machine type](https://cloud.google.com/compute/docs/machine-types) (e.g. `n2-custom-64-262144`).
+- `{machine}+{accelerator}*{count}` - Any machine and accelerator combination (e.g. `custom-8-53248+nvidia-tesla-k80*1`).
+
+#### Microsoft Azure
+
+- `{machine}` - Any [Azure VM](https://azure.microsoft.com/en-us/pricing/vm-selector) (e.g. `Standard_F8s_v2`).
+
+#### Kubernetes
+
+- `{cpu}-{memory}` - Any [CPU & memory combination](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers) (e.g. `64-256000`).
+- `{cpu}-{memory}+{accelerator}*{count}` - Any CPU, memory, & [accelerator](https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus) combination (e.g. `64-256000+nvidia-tesla-k80*1`).
+
+-> **Note** Specified resource amounts are considered **limits** (rather than **requests**).
+
+-> **Note** `{accelerator}` will be transformed into a node selector requesting `accelerator={accelerator}` and `{count}` will be configured as the **limits** count for `kubernetes.io/gpu`.
+
+## Machine Images
+
+### Generic
+
+The Iterative Provider offers some common machine images which are roughly the same
+for all supported clouds.
+
+- `ubuntu` - Official Ubuntu LTS image, currently 20.04.
+
+### Cloud-specific
+
+In addition to generic images, it's possible to specify any machine image
+supported by the underlying cloud provider.
+
+#### Amazon Web Services
+
+`{user}@{architecture}:{owner}:{name}`
+
+Fields:
+
+- `{user}` - User name; e.g. `ubuntu`.
+- `{architecture}` - Image architecture; e.g. `x86_64` or `*` for any.
+- `{owner}` - Account number of the image owner; e.g. `099720109477` or `*` for any.
+- `{name}` - Name of the image; e.g. `*ubuntu/images/hvm-ssd/ubuntu-focal-20.04*`.
+
+See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html for more information.
+
+#### Google Cloud Platform
+
+`{user}@{project}/{family}`
+
+Fields:
+
+- `{user}` - User name; e.g. `ubuntu`.
+- `{project}` - Project name; e.g. `ubuntu-os-cloud`.
+- `{family}` - Image architecture; e.g. `ubuntu-2004-lts`.
+
+See https://cloud.google.com/compute/docs/images/os-details for more information.
+
+#### Microsoft Azure
+
+`{user}@{publisher}:{offer}:{sku}:{version}`
+
+Fields:
+
+- `{user}` - User name; e.g. `ubuntu`.
+- `{publisher}` - Image publisher; e.g. `Canonical`.
+- `{offer}` - Image offer; e.g. `UbuntuServer`.
+- `{sku}` - Image SKU; e.g. `18.04-LTS`.
+- `{version}` - Image version; e.g. `latest`.
+
+See https://docs.microsoft.com/en-us/azure/virtual-machines/linux/cli-ps-findimage for more information.
+
+### Kubernetes
+
+- `{image}` - Any [container image](https://kubernetes.io/docs/concepts/containers/images/#image-names).
+
+## Cloud Regions
+
+### Generic
+
+The Iterative Provider offers some common cloud regions which are roughly the same
```

Comment:
Should this section happen before **Machine Types/Images**, since it's mentioned and linked first?

Reply:
And how about compressing the contents of these last 3 sections in `<details>` blocks? Just a thought

Reply:
Those sections don't really belong to the standard resource documentation structure, but nesting them under the **Argument Reference** section was a bit too much and removed them from the side bar.

Reply:
Perhaps `<details>` is too hidden for some users, but I acknowledge that having all that information spread on the page might be a bit too much.

Reply:
Referencing on https://github.com/iterative/terraform-provider-iterative/pull/237#issue-776178285